File: 프로그래머스/unrated/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        boolean[] area = new boolean[n+1];
        for(int i : section){
            area[i] = true;
        }
        
        int key = section.length;
        while(key != 0){
            int idx = section[key-1];
            for(int i=0; i<m && idx>=0; i++, idx--){
                if(area[idx]){
                    area[idx] = false;
                    key--;
                }
            }
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/unrated/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    var answer = 0;
    for(var i=0; i<section.length; i++) {
        answer++;
        let start = section[i];
        while(i < section.length-1 && start+m-1 >= section[i+1]) {
            i++;
        }
    }
    return answer;
}